config/config.go

```diff
@@ -493,7 +493,7 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	if config.RemoteState != nil {
 		includedConfig.RemoteState = config.RemoteState
 	}
-	if config.PreventDestroy {
+	if config.PreventDestroy != includedConfig.PreventDestroy {
```

Comment:
I just a had a realization this might not be the right solution. `PreventDestroy` is `bool`, which means it defaults to `false`, even if the user doesn't set it. That means that if the root config has `prevent_destroy = true`, but the child config _does not have `prevent_destroy` set at all_, then it'll default to `false`, and when this conditional executes, it'll override the root setting with `false`.

I think the real fix here will be to make `PreventDestroy` a pointer to a bool and to differentiate the cases where it's not set at all (`nil`) to where it's explicitly set to `true` or `false`.